### ISSUE TITLE
[release/3.1] CI failure: CertificateCreation.CertificateRequestUsageTests.AlwaysVersion3 - unable to build chain to self-signed root for signer (null)

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#76653]
I've been seeing this failure in some corefx `release/3.1` PRs. Latest example: https://github.com/dotnet/corefx/pull/43161

- Affected queues: `MacOS Build x64_Release`
- Job results: https://dev.azure.com/dnceng-public/public/_build/results?buildId=40240&view=logs&j=c0301585-8551-5e4f-e903-7e15f25b7319&t=ac085211-69c5-5ada-e2f3-cc83ef45c356
- Log file: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-43161-merge-efcbdf50df0b43b794/System.Security.Cryptography.X509Certificates.Tests/1/console.328bbb0c.log?helixlogtype=result

<details>
<summary>Callstack</summary>

```
===========================================================================================================
/private/tmp/helix/working/C5B30A64/w/B41709B1/e /private/tmp/helix/working/C5B30A64/w/B41709B1/e
  Discovering: System.Security.Cryptography.X509Certificates.Tests (method display = ClassAndMethod, method display options = None)
Run 'security unlock-keychain' to make tests runable.
  Discovered:  System.Security.Cryptography.X509Certificates.Tests (found 430 of 491 test cases)
  Starting:    System.Security.Cryptography.X509Certificates.Tests (parallel test collections = on, max threads = 6)
Warning: unable to build chain to self-signed root for signer "(null)"
    System.Security.Cryptography.X509Certificates.Tests.CertificateCreation.CertificateRequestUsageTests.AlwaysVersion3 [FAIL]
      Interop+AppleCrypto+AppleCommonCryptoCryptographicException : Unknown format in import.
      Stack Trace:
        /_/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/X509Pal.cs(161,0): at Internal.Cryptography.Pal.X509Pal.AppleX509Pal.GetCertContentType(Byte[] rawData)
        /_/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs(352,0): at System.Security.Cryptography.X509Certificates.X509Certificate2.GetCertContentType(Byte[] rawData)
        /_/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/AppleCertificatePal.cs(79,0): at Internal.Cryptography.Pal.AppleCertificatePal.FromBlob(Byte[] rawData, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)
        /_/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/CertificatePal.cs(38,0): at Internal.Cryptography.Pal.CertificatePal.FromBlob(Byte[] rawData, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)
        /_/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate.cs(58,0): at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(Byte[] data)
        /_/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs(603,0): at System.Security.Cryptography.X509Certificates.CertificateRequest.Create(X500DistinguishedName issuerName, X509SignatureGenerator generator, DateTimeOffset notBefore, DateTimeOffset notAfter, Byte[] serialNumber)
        /_/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs(314,0): at System.Security.Cryptography.X509Certificates.CertificateRequest.CreateSelfSigned(DateTimeOffset notBefore, DateTimeOffset notAfter)
        /_/src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CertificateRequestUsageTests.cs(349,0): at System.Security.Cryptography.X509Certificates.Tests.CertificateCreation.CertificateRequestUsageTests.AlwaysVersion3()
  Finished:    System.Security.Cryptography.X509Certificates.Tests
=== TEST EXECUTION SUMMARY ===
   System.Security.Cryptography.X509Certificates.Tests  Total: 755, Errors: 0, Failed: 1, Skipped: 0, Time: 19.397s
/private/tmp/helix/working/C5B30A64/w/B41709B1/e
----- end Tue Oct 4 11:17:30 PDT 2022 ----- exit code 1 ----------------------------------------------------------
```
</details>

**Comment by vcsjones** [comment]
This is basically https://github.com/dotnet/runtime/issues/40939 which was fixed in .NET 5 at https://github.com/dotnet/runtime/pull/40944.

Based on the comment at https://github.com/dotnet/runtime/pull/40944#issuecomment-677084478, if we want to fix this for 3.1 we should just disable the test for Big Sur+ (MacOS 11 or higher) as the actual fix will not meet the servicing bar at this point for .NET Core 3.1.

Would a PR to disable the test be accepted for 3.1?

**Comment by bartonjs** [comment]
The test has been disabled in the 3.1 branch and the fix won't meet the servicing bar (the scenario is basically a usage error on the part of the caller).  No further action required.
